yt_dlp/extractor/bilibili.py

```diff
@@ -139,283 +129,272 @@ class BiliBiliIE(InfoExtractor):
+                'comment_count': traverse_obj(initial_state, ('videoData', 'stat', 'reply')),
+                'uploader': traverse_obj(initial_state, ('upData', 'name')),
+                'uploader_id': traverse_obj(initial_state, ('upData', 'mid')),
+                'tags': [t['tag_name'] for t in initial_state.get('tags', []) if 'tag_name' in t],
+            }
+
+        return {
+            **info_fmt, **other_info,
+            'id': id_str,
+            'title': title,
+            'duration': float_or_none(play_info.get('timelength'), scale=1000),
+            'subtitles': subtitles,
+            'http_headers': http_headers,
+            '__post_extractor': self.extract_comments(aid),
         }
-        '''
 
-        top_level_info['__post_extractor'] = self.extract_comments(video_id)
+    def parse_old_flv_formats(self, video_id, bv_id, cid, support_formats, http_headers):
+        formats = []
+        for f in support_formats:
+            playurl = f'https://api.bilibili.com/x/player/playurl?bvid={bv_id}&cid={cid}&qn={f["quality"]}'
+            video_info_ext = self._download_json(playurl, video_id, headers=http_headers)
+            if not video_info_ext:
+                continue
+            video_info_ext = video_info_ext['data']
 
-        for entry in entries:
-            entry.update(info)
+            slices = []
+            for durl in video_info_ext['durl']:
+                slices.append({
+                    'url': durl['url'],
+                    'filesize': int_or_none(durl['size'])
+                })
+            ext = f['format']
+            if ext.startswith('flv'):
+                # flv, flv360, flv720
+                ext = 'flv'
+
+            filesize = 0
+            for s in slices:
+                if s['filesize'] is None:
+                    filesize = None
+                else:
+                    filesize += s['filesize']
 
-        if len(entries) == 1:
-            entries[0].update(top_level_info)
-            return entries[0]
+            if len(slices) == 0:
+                continue
 
-        for idx, entry in enumerate(entries):
-            entry['id'] = '%s_part%d' % (video_id, (idx + 1))
+            fmt = {
+                'url': slices[0]['url'],
+                'ext': ext,
+                'quality': f['quality'],
+                'quality_desc': f['display_desc'],
+                'height': int_or_none(f['display_desc'].rstrip('P')),
+                'vcodec': f.get('codecs'),
+                'acodec': None,
+                'entries': slices,
+                'filesize': filesize
+            }
+            formats.append(fmt)
+        return formats
 
-        return {
-            'id': str(video_id),
-            'bv_id': bv_id,
-            'title': title,
-            'description': description,
-            **info, **top_level_info
-        }
+    def rewrite_as_multi_video(self, formats, id_str, title, http_headers):
```

Comment:
It doesn't make sense for parse_old_flv_formats to return one structure and then this function to transform it. Why cant the parser return the data in correct format?

My understanding is that we don't expect any videos with differing number of slices. So that case could throw an error, greatly simplifying this.